### PR TITLE
Introduce AMReX change of stencil for Poisson solver

### PR DESCRIPTION
**Pull request by RemiLehe**
~~This reintroduces the `AMReX` changes that were in #2478/#2484 and were removed in #2494.~~

This reintroduces the `AMReX` changes that were in #2478 and were later removed (#2494). These changes essentially modify the stencil of the Poisson solver.

A later PR will then introduce the changes associated with #2484.

### CODE REVIEW

**Comment by ax3l** [comment]
cc @KZhu-ME :)

File: Regression/WarpX-GPU-tests.ini [hunk 1]
```diff
@@ -48,7 +48,7 @@ emailBody = Check https://ccse.lbl.gov/pub/GpuRegressionTesting/WarpX/ for more
 
 [AMReX]
 dir = /home/regtester/git/amrex/
-branch = 09a0ec56378f9f6a5f68f3ac025318af746432d5
+branch = 614b9ebbbf04eb5b4532029ff4a7ae40e6663042
```
Comment (WeiqunZhang) [inline Regression/WarpX-GPU-tests.ini:51]:
```suggestion
branch = 652af10ae992e2cf5c1812e0908d9b88ac22126c
```

File: Regression/WarpX-tests.ini [hunk 2]
```diff
@@ -48,7 +48,7 @@ emailBody = Check https://ccse.lbl.gov/pub/RegressionTesting/WarpX/ for more det
 
 [AMReX]
 dir = /home/regtester/AMReX_RegTesting/amrex/
-branch = 09a0ec56378f9f6a5f68f3ac025318af746432d5
+branch = 614b9ebbbf04eb5b4532029ff4a7ae40e6663042
```
Comment (WeiqunZhang) [inline Regression/WarpX-tests.ini:51]:
```suggestion
branch = 652af10ae992e2cf5c1812e0908d9b88ac22126c
```
I think this is the one we want.

File: run_test.sh [hunk 3]
```diff
@@ -51,7 +51,7 @@ echo "cd $PWD"
 
 # Clone PICSAR, AMReX and warpx-data
 git clone https://github.com/AMReX-Codes/amrex.git
-cd amrex && git checkout --detach 09a0ec56378f9f6a5f68f3ac025318af746432d5 && cd -
+cd amrex && git checkout --detach 614b9ebbbf04eb5b4532029ff4a7ae40e6663042 && cd -
```
Comment (WeiqunZhang) [inline run_test.sh:54]:
```suggestion
cd amrex && git checkout --detach 652af10ae992e2cf5c1812e0908d9b88ac22126c && cd -
```

File: Regression/WarpX-tests.ini (hunk 2 above)
Comment (RemiLehe) [reply Regression/WarpX-tests.ini:51]: Agreed. I am just introducing these changes in a two-step way, to see which one violates the CI tests.

**Comment by KZhu-ME** [comment]
@WeiqunZhang 

When we run our tests, the AMReX commit https://github.com/AMReX-Codes/amrex/commit/652af10ae992e2cf5c1812e0908d9b88ac22126c that corresponds to https://github.com/ECP-WarpX/WarpX/pull/2478 passes our tests.
But AMReX commit https://github.com/AMReX-Codes/amrex/commit/652af10ae992e2cf5c1812e0908d9b88ac22126c that corresponds to https://github.com/ECP-WarpX/WarpX/pull/2484 breaks our embedded boundary emission. 

Below are graphs of the electrostatic potential and electric field (z) in a slice along the z direction, through the middle of the wire we have. This simulation was run using https://github.com/AMReX-Codes/amrex/commit/652af10ae992e2cf5c1812e0908d9b88ac22126c. It seems that the electrostatic potential is correct, but the calculated E field incorrectly has sharp spikes around the edges of the embedded boundary.

Here is the input file I used to generate the data for these plots. 
[inputs2d.txt](https://github.com/ECP-WarpX/WarpX/files/7444503/inputs2d.txt)

Ez_step0
phi_step0

**Comment by RemiLehe** [comment]
Many tests fail because of benchmark differences. All these tests involve the electrostatic solver:
`Plasma_lens`
`Python_prev_positions`
`RepellingParticles`
`background_mcc`
`space_charge_initialization_2d`
We think that this is because of the modified discretized stencil of the Laplacian operator, that was introduced in https://github.com/AMReX-Codes/amrex/pull/2416. I will therefore reset the benchmark for these tests.

Some additional test fail because of the Python comparison with theory:
`relativistic_space_charge_initialization`
`space_charge_initialization`.
When inspecting the fields manually for `relativistic_space_charge_initialization`, the fields still look physical:
index
I will therefore increase the tolerance of this test.

Changes are similar for the test `space_charge_initialization`. I will therefore also increase the tolerance in this case:
Screen Shot 2021-11-01 at 11 01 40 AM

File: run_test.sh (hunk 3 above)
Comment (ax3l) [reply run_test.sh:54]:
```suggestion
cd amrex && git checkout --detach 3add1f2824c5a4f33e77abc612da8a52b3f5131b && cd -
```

File: cmake/dependencies/AMReX.cmake [hunk 4]
```diff
@@ -239,7 +239,7 @@ set(WarpX_amrex_src ""
 set(WarpX_amrex_repo "https://github.com/AMReX-Codes/amrex.git"
     CACHE STRING
     "Repository URI to pull and build AMReX from if(WarpX_amrex_internal)")
-set(WarpX_amrex_branch "09a0ec56378f9f6a5f68f3ac025318af746432d5"
+set(WarpX_amrex_branch "614b9ebbbf04eb5b4532029ff4a7ae40e6663042"
```
Comment (ax3l) [inline cmake/dependencies/AMReX.cmake:242]:
```suggestion
set(WarpX_amrex_branch "3add1f2824c5a4f33e77abc612da8a52b3f5131b"
```

File: Regression/WarpX-tests.ini (hunk 2 above)
Comment (ax3l) [reply Regression/WarpX-tests.ini:51]:
```suggestion
branch = 3add1f2824c5a4f33e77abc612da8a52b3f5131b
```

File: Regression/WarpX-GPU-tests.ini (hunk 1 above)
Comment (ax3l) [reply Regression/WarpX-GPU-tests.ini:51]:
```suggestion
branch = 3add1f2824c5a4f33e77abc612da8a52b3f5131b
```

**Comment by ax3l** [comment]
@RemiLehe let's update AMReX once more, fix for `compGrad` landed in https://github.com/AMReX-Codes/amrex/pull/2453 aka `3add1f2824c5a4f33e77abc612da8a52b3f5131b`

**Review by ax3l — APPROVED** [review]
Thank you - as discussed: if this passes we add another PR that updated to the latest, 21.11+bugfix AMReX to address the `compGrad` issue.

**Pull request merged**